plugins/repository-hdfs tests are failing when built by JDK 11

Feature branch: `feature/31230_gradle48_jdk11`

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+31230_gradle48_jdk11+matrix-java-feature-branch/5/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/#showFailuresLink

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+31230_gradle48_jdk11+matrix-java-feature-branch/5/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java10,nodes=virtual&&linux/#showFailuresLink

```
stack_trace" : "RepositoryException[[test_snapshot_repository_ro] failed to create repository]; nested: UncheckedIOException[Could not retrieve the current user information]; nested: IOException[failure to login: javax.security.auth.login.LoginException: Security Exception]; nested: LoginException[Security Exception]; nested: SecurityException;
	at org.elasticsearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:389)
	at org.elasticsearch.repositories.RepositoriesService.registerRepository(RepositoriesService.java:357)
	at org.elasticsearch.repositories.RepositoriesService.access$100(RepositoriesService.java:55)
	at org.elasticsearch.repositories.RepositoriesService$1.execute(RepositoriesService.java:108)
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:634)
	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:266)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:196)
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:131)
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:623)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.io.UncheckedIOException: Could not retrieve the current user information
	at org.elasticsearch.repositories.hdfs.HdfsRepository.login(HdfsRepository.java:194)
	at org.elasticsearch.repositories.hdfs.HdfsRepository.createBlobstore(HdfsRepository.java:123)
	at org.elasticsearch.repositories.hdfs.HdfsRepository.lambda$doStart$0(HdfsRepository.java:104)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.elasticsearch.repositories.hdfs.HdfsRepository.doStart(HdfsRepository.java:104)
	at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:66)
	at org.elasticsearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:385)
	... 16 more
Caused by: java.io.IOException: failure to login: javax.security.auth.login.LoginException: Security Exception
	at org.apache.hadoop.security.UserGroupInformation.loginUserFromSubject(UserGroupInformation.java:877)
	at org.apache.hadoop.security.UserGroupInformation.getLoginUser(UserGroupInformation.java:802)
	at org.apache.hadoop.security.UserGroupInformation.getCurrentUser(UserGroupInformation.java:675)
	at org.elasticsearch.repositories.hdfs.HdfsRepository.login(HdfsRepository.java:192)
	... 22 more
Caused by: javax.security.auth.login.LoginException: Security Exception
	at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:805)
	at java.base/javax.security.auth.login.LoginContext.access$000(LoginContext.java:194)
	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:665)
	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:663)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:663)
	at java.base/javax.security.auth.login.LoginContext.login(LoginContext.java:574)
	at org.apache.hadoop.security.UserGroupInformation.loginUserFromSubject(UserGroupInformation.java:840)
	... 25 more
Caused by: java.lang.SecurityException
	at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:806)
	... 32 more
```

--- a/plugins/repository-hdfs/build.gradle
+++ b/plugins/repository-hdfs/build.gradle
@@ -1,4 +1,4 @@
-1/*
+/*
  * Licensed to Elasticsearch under one or more contributor
  * license agreements. See the NOTICE file distributed with
  * this work for additional information regarding copyright
@@ -212,25 +212,6 @@ RestIntegTestTask integTestSecure = project.tasks.create('integTestSecure', Rest
 // Create a Integration Test suite just for HA related security based tests
 RestIntegTestTask integTestSecureHa = project.tasks.create('integTestSecureHa', RestIntegTestTask.class) {
   description = "Runs rest tests against an elasticsearch cluster with HDFS configured with HA Namenode and secured by MIT Kerberos."
-}
-
-if (rootProject.ext.compilerJavaVersion.isJava11()) {
-  // TODO remove when: https://github.com/elastic/elasticsearch/issues/31498
-  integTestRunner {
-    systemProperty 'tests.rest.blacklist', [
-            'hdfs_repository/30_snapshot/take snapshot',
-            'hdfs_repository/40_restore/Create a snapshot and then restore it',
-            'hdfs_repository/20_repository_verify/HDFS Repository Verify',
-            'hdfs_repository/30_snapshot_get/Get a snapshot',
-            'hdfs_repository/20_repository_create/HDFS Repository Creation',
-            'hdfs_repository/20_repository_delete/HDFS Delete Repository',
-            'hdfs_repository/30_snapshot_readonly/Get a snapshot - readonly',
-    ].join(',')
-  }
-}
-if (rootProject.ext.runtimeJavaVersion.isJava11() || rootProject.ext.compilerJavaVersion.isJava11()) {
-  // TODO remove when: https://github.com/elastic/elasticsearch/issues/31498
-  integTestHa.enabled = false
 }
 
 // Determine HDFS Fixture compatibility for the current build environment.

--- a/plugins/repository-hdfs/src/main/plugin-metadata/plugin-security.policy
+++ b/plugins/repository-hdfs/src/main/plugin-metadata/plugin-security.policy
@@ -61,6 +61,7 @@ grant {
 
   // Hadoop depends on OS level user information for simple authentication
   // Unix: UnixLoginModule: com.sun.security.auth.module.UnixSystem.UnixSystem init
+  permission java.lang.RuntimePermission "loadLibrary.jaas";
   permission java.lang.RuntimePermission "loadLibrary.jaas_unix";
   // Windows: NTLoginModule: com.sun.security.auth.module.NTSystem.loadNative
   permission java.lang.RuntimePermission "loadLibrary.jaas_nt";

